python/snet_deframer.py

```diff
@@ -165,9 +166,22 @@ def handle_msg(self, msg_pmt):
             return
 
         pdu = np.packbits(pdu_bytes)
+        pdu_parsed = PDUFrameHeader.parse(pdu)
+
+        if self.verbose:
+            print(pdu_parsed)
+
+        # as output, it's more useful to know the payload and the kind of
+        # payload
         pdu_tags = pmt.make_dict()
         pdu_tags = pmt.dict_add(
-            pdu_tags, pmt.intern('SNET SrcId'), pmt.from_long(hdr.SrcId))
+            pdu_tags, pmt.intern('UserData'), pmt.to_pmt(pdu_parsed.UserData))
+        pdu_tags = pmt.dict_add(
+            pdu_tags, pmt.intern('FCIDSub'), pmt.to_pmt(pdu_parsed.FCIDSub))
+        pdu_tags = pmt.dict_add(
+            pdu_tags, pmt.intern('FCIDMajor'),
+            pmt.to_pmt(pdu_parsed.FCIDMajor))
+
         self.message_port_pub(
             pmt.intern('out'),
-            pmt.cons(pdu_tags, pmt.init_u8vector(len(pdu), pdu)))
+            pdu_tags)
```

Comment:
Deframers in gr-satellites should output the frame as a GNU Radio PDU, which means a `pmt.cons()` of some metadata (not obtainable from the frame contents) and the frame bytes as a PMT u8vector). Any parsing of the frame into fields should be done in a telemetry parser. Sometimes it's difficult to say what a "frame" exactly should be (should it include the CRC?, should it include the length field if any?, etc.), but for S-NET I think that `pdu` is the frame. Therefore, the original code seems fine to me.

Reply:
oh, that is a good point and I totally see how my changes are bias towards what, I feel is important, instead of a more generic approach that works for the overall. Thanks for the notes about the output to be a GNU Radio PDU, I definitely downgraded the importance of it.

Aside from this, do you think in the future, having parameters like this https://github.com/daniestevez/gr-satellites/blob/09b0d84ada7a168075cea751cdc08d589e6198f3/grc/satellites_snet_deframer.block.yml#L16 the return specific sections of the pdu make sense for the concept of deframer, or will it be always better to output everything and let an extra block or application parse the bits of interest?

Reply:
I don't think that makes sense for a deframer. From the documentation:
> Deframers. Deframers implement the lower layer protocols related to frame boundary detection, descrambling, deinterleaving, FEC, error checking with a CRC code, etc. The output of a deframer are PDUs with the frames. Some examples are an AX.25 deframer and a CCSDS concatenated code deframer.

Reply:
thanks for the clarification, I rolled back this change.